src/modules/commander/state_machine_helper.cpp

```diff
@@ -1102,6 +1060,93 @@ bool set_nav_state(struct vehicle_status_s *status, struct commander_state_s *in
 	return status->nav_state != nav_state_old;
 }
 
+void set_rc_loss_nav_state(struct vehicle_status_s* status,
+						   struct actuator_armed_s* armed,
+						   status_flags_s* status_flags,
+						   const link_loss_actions_t link_loss_act)
+{
+	set_link_loss_nav_state(status, armed, status_flags, link_loss_act, vehicle_status_s::NAVIGATION_STATE_AUTO_RCRECOVER);
+}
+
+void set_data_link_loss_nav_state(struct vehicle_status_s* status,
+								  struct actuator_armed_s* armed,
+								  status_flags_s* status_flags,
+								  const link_loss_actions_t link_loss_act)
+{
+	set_link_loss_nav_state(status, armed, status_flags, link_loss_act, vehicle_status_s::NAVIGATION_STATE_AUTO_RTGS);
+}
+
+void set_link_loss_nav_state(struct vehicle_status_s* status,
```

Comment:
What exactly looks complicated here? This is mostly port of the link loss code from the navigator to commander + 2 new action types.
I have introduced enum for actions instead of using magic numbers and merged handlers for RC and data loss as the are the same except 1 nav state.